common/utils/net.go

```diff
@@ -0,0 +1,197 @@
+package utils
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"golang.org/x/net/http/httpproxy"
+)
+
+// default timeout 30 seconds
+var defaultTimeout = 30 * time.Second
+
+type VjbNet struct {
+	Client          *http.Client
+	timeout         time.Duration
+	Insecure        bool
+	HTTPProxy       string
+	HTTPSProxy      string
+	NoProxy         string
+	UseProxyFromEnv bool
+	proxyCfg        *httpproxy.Config
+}
+
+func (v *VjbNet) getNetTransport(tlsConfig *tls.Config) *http.Transport {
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	if v.Insecure {
+		transport.TLSClientConfig.InsecureSkipVerify = true
+	}
+	// Configure proxy behavior using httpproxy.Config, which correctly
+	// handles HTTP(S)_PROXY and NO_PROXY semantics including lists and
+	// domain / CIDR matching.
+	if v.UseProxyFromEnv || v.HTTPProxy != "" || v.HTTPSProxy != "" || v.NoProxy != "" {
+		if v.proxyCfg == nil {
+			v.proxyCfg = httpproxy.FromEnvironment()
+		}
+		if v.HTTPProxy != "" {
+			v.proxyCfg.HTTPProxy = v.HTTPProxy
+		}
+		if v.HTTPSProxy != "" {
+			v.proxyCfg.HTTPSProxy = v.HTTPSProxy
+		}
+		if v.NoProxy != "" {
+			v.proxyCfg.NoProxy = v.NoProxy
+		}
+
+		transport.Proxy = func(req *http.Request) (*url.URL, error) {
+			proxyURL, err := v.proxyCfg.ProxyFunc()(req.URL)
+			if err != nil {
+				return nil, err
+			}
+			// Preserve existing logging behavior.
+			if proxyURL != nil {
+				fmt.Printf("Proxy config: HTTPProxy=%s, HTTPSProxy=%s, NoProxy=%s\n",
+					v.proxyCfg.HTTPProxy, v.proxyCfg.HTTPSProxy, v.proxyCfg.NoProxy)
+				fmt.Printf("VjbNet proxy config: HTTPProxy=%s, HTTPSProxy=%s, NoProxy=%s\n",
+					v.HTTPProxy, v.HTTPSProxy, v.NoProxy)
+			}
+			return proxyURL, nil
+		}
+	}
+	return transport
+}
+
+// CreateHTTPClient creates an HTTP client with TLS configuration and retry logic
+func (v *VjbNet) CreateHTTPClient() error {
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+	if v.Insecure {
+		tlsConfig.InsecureSkipVerify = true
+	}
+	transport := v.getNetTransport(tlsConfig)
+
+	if v.UseProxyFromEnv {
+		transport.Proxy = http.ProxyFromEnvironment
+	}
```

Comment:
we may need the proxy setting which are explicitly set.
Like if we can always do
```
transport.Proxy = v.ProxyConfig
```